corehq/sql_db/routers.py

```diff
@@ -115,7 +116,14 @@ def db_for_read_write(model, write=True):
             return plproxy_config.proxy_db
         return DEFAULT_DB_ALIAS
     if app_label == FORM_PROCESSOR_APP:
-        return plproxy_config.proxy_db
+        if hints.get('plproxy_read'):
+            return plproxy_config.proxy_db
+        if 'using' in hints:
+            return hints['using']
+        if 'partition_value' in hints:
+            from corehq.sql_db.util import get_db_alias_for_partitioned_doc
+            return get_db_alias_for_partitioned_doc(hints['partition_value'])
```

Comment:
Should this enforce that only one of 'plproxy_read', 'using', or 'partition_value' is set? Looks like if multiple are passed in all but one will be ignored. I'm not familiar with the `hints` parameter, so if there's a convention I'm missing, would love to be educated!

Reply:
good idea: 38a9a91a1dc

Hints are a mechanism that Django provides to give the DB router additional information to make routing decisions on: https://docs.djangoproject.com/en/2.2/topics/db/multi-db/#hints